ir/src/transition_stmts/graph.rs

```diff
@@ -200,6 +205,28 @@ impl AlgebraicGraph {
                 )));
                 Ok((trace_segment, node_index))
             }
+            IdentifierType::TransitionVariable(transition_variable) => {
+                if let TransitionVariableType::Scalar(expr) = transition_variable.value() {
+                    if let Some((trace_segment, node_index)) =
+                        variable_roots.get(&VariableValue::Scalar(ident.to_string()))
+                    {
+                        Ok((*trace_segment, *node_index))
+                    } else {
+                        let (trace_segment, node_index) =
+                            self.insert_expr(symbol_table, expr.clone(), variable_roots)?;
+                        variable_roots.insert(
+                            VariableValue::Scalar(ident.to_string()),
+                            (trace_segment, node_index),
+                        );
+                        Ok((trace_segment, node_index))
+                    }
+                } else {
+                    Err(SemanticError::InvalidUsage(format!(
+                        "Identifier {} was declared as a {} which is not a supported type.",
+                        ident, elem_type
+                    )))
+                }
+            }
```

Comment:
There's quite a bit of similarity/near-redundancy between this, `insert_vector_access`, and `insert_matrix_access, so I think there's probably a better way. Can you take a look at refactoring a bit or ping me if you want to discuss?

Reply:
I agree with you that there is some redundancy but I can't think of a way to reduce it much here.

Reply:
Ok, let's leave it for now and revisit in the future once the tests are in place.